modules/openy_gc_auth/src/GCIdentityProviderPluginBase.php

```diff
@@ -0,0 +1,157 @@
+ * @see \Drupal\openy_gc_auth\Annotation\GCIdentityProvider
+ * @see plugin_api
+ */
+abstract class GCIdentityProviderPluginBase extends PluginBase implements GCIdentityProviderInterface, ContainerFactoryPluginInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * The configuration factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configFactory = $config;
+    // We use pre-saved configuration here.
+    $configuration = $this->configFactory->get($this->getConfigName())->get();
+    $this->setConfiguration($configuration);
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static($configuration, $plugin_id, $plugin_definition, $container->get('config.factory'), $container->get('entity_type.manager'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getId() {
+    return $this->pluginDefinition['id'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLabel() {
+    return $this->pluginDefinition['label'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfigName() {
+    return $this->pluginDefinition['config'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  private function baseConfigurationDefaults():array {
+    return [
+      // TODO: add here configs that can be similar for all instances.
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration():array {
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfiguration() {
+    return $this->configuration;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setConfiguration(array $configuration) {
+    $this->configuration = NestedArray::mergeDeep(
+      $this->baseConfigurationDefaults(),
+      $this->defaultConfiguration(),
+      $configuration
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    // $config = $this->getConfiguration();
+    $form['admin_label'] = [
+      '#type' => 'page_title',
+      '#title' => $this->getLabel(),
+    ];
+    // TODO: add here fields that can be similar for all instances.
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state) {
+    // This method not required.
```

Comment:
so maybe we can get rid of it?

Reply:
This plugin used **PluginFormInterface** and **validateConfigurationForm** method required in extended classes. Most of the plugins without this validation, so to avoid empty method in each plugin instance it was implemented in base class